File: app/src/main/java/com/kaartiikvjn/empayarbatik/utils/ImageHelper.java
package com.kaartiikvjn.empayarbatik.utils;

import android.content.Context;
import android.util.Log;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;
import com.kaartiikvjn.empayarbatik.R;

public class ImageHelper {
    private Context context;
    private StorageReference bucketReference;
    private static final String TAG = "ImageHelper";

    public ImageHelper(Context context) {
        this.context = context;
        bucketReference = FirebaseStorage.getInstance().getReference();
    }

    public void loadCloudImage(String url, ImageView target) {
        bucketReference.child(url).getDownloadUrl().addOnSuccessListener(uri -> {
            Glide.with(context).load(uri)
                    .placeholder(R.drawable.photo_loading)
                    .error(R.drawable.photo_unavailable)
                    .into(target);
        }).addOnFailureListener(e -> Log.d(TAG, "loadCloudImage: " + e.getMessage()));
    }
}

File: app/src/main/java/com/kaartiikvjn/empayarbatik/utils/ImageHelper.kt
package com.kaartiikvjn.empayarbatik.utils

import android.content.Context
import android.net.Uri
import android.util.Log
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.google.firebase.storage.FirebaseStorage
import com.google.firebase.storage.StorageReference
import com.kaartiikvjn.empayarbatik.R

class ImageHelper(private val context: Context) {
    companion object {
        private const val TAG = "ImageHelper"
    }

    var bucketReference: StorageReference? = null


    fun loadImage(url: String?, imageView: ImageView?) {
        bucketReference!!.child(url!!).downloadUrl.addOnSuccessListener { uri: Uri? ->
            Glide.with(
                context
            )
                .load(uri).
                placeholder(R.drawable.photo_loading)
                .error(R.drawable.photo_unavailable)
                .into(imageView!!)
        }.addOnFailureListener { e: Exception ->
            Log.d(
                TAG,
                "loadImage: " + e.message + "\n path:$url"
            )
        }
    }

    init {
        bucketReference = FirebaseStorage.getInstance().reference
    }
}